src/transaction.ts

```diff
@@ -61,25 +59,63 @@ export function assembleTransaction(
     extraSigners: raw.extraSigners,
   });
 
-  // apply the auth from the simulation to the invokeHostFunction op's props
-  const invokeOp: Operation.InvokeHostFunction = raw.operations[0];
-  txnBuilder.addOperation(
-    Operation.invokeHostFunction({
-      func: invokeOp.func,
-      auth: (invokeOp.auth ?? []).concat(
-        simulation.results[0].auth?.map((a: string) =>
-          xdr.SorobanAuthorizationEntry.fromXDR(a, "base64")
-        ) ?? []
-      ),
-    })
-  );
+  switch (raw.operations[0].type) {
+    case "invokeHostFunction":
+      const invokeOp: Operation.InvokeHostFunction = raw.operations[0];
+      txnBuilder.addOperation(
+        Operation.invokeHostFunction({
+          source: invokeOp.source,
+          func: invokeOp.func,
+          // apply the auth from the simulation
+          auth: (invokeOp.auth ?? []).concat(
+            simulation.results[0].auth?.map((a: string) =>
+              xdr.SorobanAuthorizationEntry.fromXDR(a, "base64")
+            ) ?? []
+          ),
+        })
+      );
+      break;
+
+    case "bumpFootprintExpiration":
+      const bumpOp: Operation.BumpFootprintExpiration = raw.operations[0];
+      txnBuilder.addOperation(
```

Comment:
since this and restore case are not merging any data from simulation, can it reduce to `txnBuilder.addOperation(raw.operations[0])`, or I may be missing a subtlety.

Reply:
I thought so, too, but kept getting some TypeScript errors... tried it again and it worked :facepalm: Great call, thank you!